scripts/update_and_rebuild_mfem.sh

```diff
@@ -0,0 +1,121 @@
+  echo
+  echo "-h | --help              Display this message and list of available libmesh options"
+  echo "--skip-submodule-update  Do not update the libMesh submodule, use the current version"
+  echo
+  echo "Influential variables"
+  echo "CONDUIT_SRC_DIR          Path to conduit; default: ../framework/contrib/conduit from submodule"
+  echo "LIBMESH_DIR              Path to libmesh (for netcdf); default ../libmesh/installed"
+  echo "MFEM_DIR                 MFEM install prefix; default: ../framework/contrib/mfem/installed"
+  echo "MFEM_SRC_DIR             Path to MFEM source; default: ../framework/contrib/mfem from submodule"
+  echo "PETSC_ARCH               PETSc arch; default: arch-moose if PETSC_DIR not set"
+  echo "PETSC_DIR                Path to PETSc install; default: ../petsc"
+  exit 0
+fi
+
+set -e
+
+if [ -n "$MFEM_SRC_DIR" ]; then
+  skip_sub_update=1
+else
+  MFEM_SRC_DIR="$(realpath -m "${SCRIPT_DIR}"/../framework/contrib/mfem)"
+fi
+MFEM_BUILD_DIR="${MFEM_SRC_DIR}/build"
+if [ -n "$MFEM_DIR" ]; then
+  echo "INFO: MFEM_DIR set - overriding default installed path"
+  echo "INFO: No cleaning will be done in specified path"
+else
+  MFEM_DIR="${MFEM_SRC_DIR}/installed"
+  rm -rf "$MFEM_DIR"
+fi
+
+CONDUIT_BUILD_DIR="${CONDUIT_SRC_DIR}/build"
+LIBMESH_DIR=${LIBMESH_DIR:-$(realpath -m "${SCRIPT_DIR}/../libmesh/installed")}
+if [ -z "$PETSC_DIR" ]; then
+  PETSC_DIR=$(realpath -m "${SCRIPT_DIR}/../petsc/arch-moose")
+fi
+
+if [ -z "$skip_sub_update" ]; then
+  cd "${SCRIPT_DIR}/.."
+  git submodule update --init --checkout framework/contrib/conduit
+  git submodule update --init --checkout framework/contrib/mfem
+  cd framework/contrib/conduit
+  git submodule update --init
+fi
+
+# Clean up old builds
+rm -rf "$CONDUIT_BUILD_DIR" "$MFEM_BUILD_DIR"
+
+# Build and install conduit
+mkdir -p "$CONDUIT_BUILD_DIR"
+cd "$CONDUIT_BUILD_DIR"
+cmake ../src -DCMAKE_INSTALL_PREFIX="$MFEM_DIR"
+make -j "${MOOSE_JOBS:-4}"
+make install
+
+# Build and install mfem
+mkdir -p "$MFEM_BUILD_DIR"
+cd "$MFEM_BUILD_DIR"
+cmake .. \
+    -DCMAKE_POSITION_INDEPENDENT_CODE=YES \
+    -DMFEM_USE_OPENMP=NO \
+    -DMFEM_THREAD_SAFE=NO \
+    -DHYPRE_DIR="$PETSC_DIR" \
+    -DPETSC_DIR="$PETSC_DIR" \
+    -DMFEM_USE_MPI=YES \
+    -DMFEM_USE_METIS=YES \
+    -DMFEM_USE_METIS_5=YES \
+    -DMETIS_DIR="$PETSC_DIR" \
+    -DParMETIS_DIR="$PETSC_DIR" \
+    -DMFEM_USE_SUPERLU=YES \
+    -DSuperLUDist_DIR="$PETSC_DIR" \
+    -DBUILD_SHARED_LIBS=ON \
+    -DHDF5_DIR="$PETSC_DIR" \
+    -DBLAS_DIR="$PETSC_DIR" \
+    -DMFEM_ENABLE_EXAMPLES=yes \
+    -DMFEM_ENABLE_MINIAPPS=yes \
+    -DBLAS_LIBRARIES="$PETSC_DIR/lib/libfblas.a" \
+    -DLAPACK_LIBRARIES="$PETSC_DIR/lib/libflapack.a" \
+    -DBLAS_INCLUDE_DIRS="$PETSC_DIR/include" \
+    -DLAPACK_INCLUDE_DIRS="$PETSC_DIR/include" \
+    -DCMAKE_INSTALL_PREFIX="$MFEM_DIR" \
+    -DMFEM_USE_PETSC=YES \
+    -DPETSC_DIR="$PETSC_DIR" \
```

Comment:
Still need to resolve https://github.com/idaholab/moose/pull/29635#discussion_r1905820665

In-tree install of petsc with a PETSC_ARCH requires PETSC_ARCH, and requires all of the PETSC_DIRs for the other deps to be PETSC_DIR/PETSC_ARCH

Reply:
I think I've resolved at least part of this